tests/unittests/tests-pkt/tests-pkt.c

```diff
@@ -106,6 +109,25 @@ static void test_pkt_count__null(void)
     TEST_ASSERT_EQUAL_INT(0, gnrc_pkt_count(NULL));
 }
 
+static void test_pkt_type(void)
+{
+    /* init packet snips */
+    gnrc_pktsnip_t snip1 = _INIT_ELEM_STATIC_TYPE(GNRC_NETTYPE_UNDEF, NULL);
+    gnrc_pktsnip_t snip2 = _INIT_ELEM_STATIC_TYPE(GNRC_NETTYPE_TEST, &snip1);
+    gnrc_pktsnip_t snip3 = _INIT_ELEM_STATIC_TYPE(GNRC_NETTYPE_IPV6, &snip2);
+    /* successfull searches */
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF,
+                          gnrc_pktsnip_search_type(&snip3, GNRC_NETTYPE_UNDEF)->type);
```

Comment:
Maybe check, if the function returns a non-NULL value to prevent (for the tester) unclear crashes in an error case.